src/main/java/com/google/sps/servlets/FilterServlet.java

```diff
@@ -0,0 +1,194 @@
+import java.util.ArrayList;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+
+@WebServlet("/filter")
+public class FilterServlet extends HttpServlet {
+  
+  private static final int COMPARISON_LIMIT = 51;
+  private static final int NEEDED_MATCHES = 5;
+
+  class Summary {
+    private final long id;
+    private final String name;
+    private final String imgUrl;
+    private final String type;
+    private final String tone;
+    private final String vegan;
+    private final String brand;
+    private final String productUrl;
+    private final long cost;
+
+    Summary(long id, String name, String imgUrl, String type, String tone, 
+            String vegan, String brand, String productUrl, long cost) {
+      this.id = id;
+      this.name = name;
+      this.imgUrl = imgUrl;
+      this.type = type;
+      this.tone = tone;
+      this.vegan = vegan;
+      this.brand = brand;
+      this.productUrl = productUrl;
+      this.cost = cost;
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+    throws IOException {
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ArrayList<Summary> summaries = new ArrayList<>();
+    ArrayList<String> types = new ArrayList<>();
+    ArrayList<String> tones = new ArrayList<>();
+    ArrayList<String> vegan = new ArrayList<>();
+    ArrayList<String> brands = new ArrayList<>();
+    ArrayList<Long> costs = new ArrayList<>();
+
+    int count = 0;
+
+    String typeFilter = request.getParameter("type");
+    String toneFilter = request.getParameter("tone");
+    String veganFilter = request.getParameter("vegan");
+    String brandFilter = request.getParameter("brand");
+    String costFilter = request.getParameter("price");
+
+    types = parseFilters(typeFilter);
+
+    tones = parseFilters(toneFilter);
+
+    vegan = parseFilters(veganFilter);
+
+    brands = parseFilters(brandFilter);
+
+    int curr = 0;
+
+    if (!costFilter.equals("None")) {
+      
+      String value;
+      long number;
+     
+      for (int i = 0; i < costFilter.length(); i++) {
+        if (costFilter.charAt(i) == '|') {
+          value = costFilter.substring(curr, i);
+          curr = i + 1;
+          number = Long.parseLong(value);
+
+          costs.add(number);
+        }
+      }
```

Comment:
Can be replaced by String.split()?

Reply:
I tried using it, but there is a small problem. String.split creates an empty string at the end due to the way that I formatted the string to be parsed if less than the maximum checkboxes are marked.